File: teste/src/main/java/com/br/teste/projeto/services/EmpresaService.java
package com.br.teste.projeto.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;

import com.br.teste.projeto.models.Empresa;
import com.br.teste.projeto.repository.IEmpresaRepository;

@Service
public class EmpresaService {

	@Autowired
	private IEmpresaRepository empresaRepository;
	
	/**
	 * Persiste o dados de uma empresa no banco.<br>
	 * Atributos de preenchimento <b>txNome</b>.
	 * @param Empresa Objeto
	 * @return Empresa preenchido com a empresa persistida
	 */
	public Empresa save(Empresa pEmpresa) {
		return empresaRepository.save(pEmpresa);
	}
	
	
	
	/**
	 * Recupera as empresas cadastradas no banco.<br>
	 * Atributos de preenchimento <b>Nenhum</b>.
	 * @param Empresa Objeto
	 * @return lista Empresa que estao cadastradas no banco sem filtro
	 */
	public List<Empresa> findAll() {
		return empresaRepository.findAll();
	}
	
	/**
	 * Exclui uma empresa do banco.<br>
	 * Atributos de preenchimento <b>Empresa empresa</b>.
	 * @param Empresa Objeto
	 * @return a mensagem de sucesso ou n sucesso
	 */
	public boolean delete(@PathVariable Long codId) {
		Empresa empresa = empresaRepository.findById(codId).get();
		
		if(empresa == null) {
			 return false;
		}
		empresaRepository.delete(empresa);
		return true;
	}
	
}

File: teste-app/src/app/services/empresa.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Empresa } from '../models/empresa';

@Injectable({
  providedIn: 'root'
})
export class EmpresaService {

  pathApi = 'http://127.0.0.1:8080/empresa';

  constructor(
    private http: HttpClient
  ) { }

  // METODO RESPONSAVEL POR PERSISTIR OBJETO NO BANCO
  save(pEmpresa: Empresa) {
    return this.http.post<Empresa>(`${this.pathApi}`, pEmpresa);
  }

  // RECUPERA TODAS EMPRESAS DO BANCO SEM FILTRO
  findAll(): Observable<Empresa[]> {
    return this.http.get<Empresa[]>(`${this.pathApi}`);
  }

  // DELETA UMA EMPRESA
  delete(codId: number) {
    return this.http.delete(`${this.pathApi}/${codId}`);
  }
}
